rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -25,10 +26,28 @@
 
 from sawtooth_rest_api.protobuf.validator_pb2 import Message
 
-
 LOGGER = logging.getLogger(__name__)
 
 
+class _Backoff:
+    """Implements a simple exponential backoff mechanism.
+    """
+
+    def __init__(self):
+        self.num_retries_left = 3
```

Comment:
Is this decremented somewhere?

Reply:
This value should be a parameter, as well.  It can be a keyword param, with a default value.